### ISSUE TITLE
Nested merge not works like expected if data input is empty

### ISSUE THREAD

**Issue by lublak** [n8n-io/n8n#1286]
**Describe the bug**
Nested merge nodes not works well.

![grafik](https://user-images.githubusercontent.com/44057030/103352387-6b4a7380-4aa6-11eb-813e-c0dfdb0a4b9d.png)

After press Execute Workflow:

![grafik](https://user-images.githubusercontent.com/44057030/103352411-78fff900-4aa6-11eb-9011-d774edc409e4.png)

After press manual execute on Merge 2:
![grafik](https://user-images.githubusercontent.com/44057030/103352522-b2d0ff80-4aa6-11eb-986a-f90f336c6bd9.png)

than on Merge:

![grafik](https://user-images.githubusercontent.com/44057030/103352547-c4b2a280-4aa6-11eb-8b31-6327e45a6af7.png)

**To Reproduce**

Nest merge nodes.

**Expected behavior**

Execute Merge recusive.

**Environment (please complete the following information):**
 - OS: Windows 10
 - n8n Version: 0.100.0 
 - Node.js Version: v14.15.1

**Additional context**

If i press Merge three times it also works:

![grafik](https://user-images.githubusercontent.com/44057030/103352792-7eaa0e80-4aa7-11eb-8c87-aca5fdfedbed.png)

**Comment by janober** [comment]
Ah very interesting. If that happens like you describe it really looks like a bug. A temporary workaround would be to connect the "Start" node also to "Read folder 2". It should then work fine.

**Comment by lublak** [comment]
Yes this works fine as workaround :)

**Comment by janober** [comment]
Can you please create a simple workflow with which we can reproduce the bug you are experiencing? I just tried with the following (also with more nodes) but always worked totally fine for me.
![Screenshot from 2020-12-30 14-31-37](https://user-images.githubusercontent.com/6249596/103354417-bf0b8b80-4aab-11eb-8f68-88b6ff1456d3.png)

```json
{
  "nodes": [
    {
      "parameters": {},
      "name": "Start",
      "type": "n8n-nodes-base.start",
      "typeVersion": 1,
      "position": [
        250,
        300
      ]
    },
    {
      "parameters": {
        "values": {
          "number": [
            {
              "name": "number0"
            }
          ],
          "string": [
            {
              "name": "key",
              "value": "a"
            }
          ]
        },
        "options": {}
      },
      "name": "Set",
      "type": "n8n-nodes-base.set",
      "typeVersion": 1,
      "position": [
        450,
        300
      ]
    },
    {
      "parameters": {
        "values": {
          "number": [
            {
              "name": "number1",
              "value": 1
            }
          ],
          "string": [
            {
              "name": "key",
              "value": "a"
            }
          ]
        },
        "options": {}
      },
      "name": "Set1",
      "type": "n8n-nodes-base.set",
      "typeVersion": 1,
      "position": [
        450,
        450
      ]
    },
    {
      "parameters": {
        "values": {
          "number": [
            {
              "name": "number2",
              "value": 2
            }
          ],
          "string": [
            {
              "name": "key",
              "value": "a"
            }
          ]
        },
        "options": {}
      },
      "name": "Set2",
      "type": "n8n-nodes-base.set",
      "typeVersion": 1,
      "position": [
        450,
        600
      ]
    },
    {
      "parameters": {
        "mode": "mergeByKey",
        "propertyName1": "key",
        "propertyName2": "key"
      },
      "name": "Merge",
      "type": "n8n-nodes-base.merge",
      "typeVersion": 1,
      "position": [
        700,
        450
      ]
    },
    {
      "parameters": {
        "mode": "mergeByKey",
        "propertyName1": "key",
        "propertyName2": "key"
      },
      "name": "Merge1",
      "type": "n8n-nodes-base.merge",
      "typeVersion": 1,
      "position": [
        900,
        300
      ]
    }
  ],
  "connections": {
    "Start": {
      "main": [
        [
          {
            "node": "Set",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "Set": {
      "main": [
        [
          {
            "node": "Merge1",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "Set1": {
      "main": [
        [
          {
            "node": "Merge",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "Set2": {
      "main": [
        [
          {
            "node": "Merge",
            "type": "main",
            "index": 1
          }
        ]
      ]
    },
    "Merge": {
      "main": [
        [
          {
            "node": "Merge1",
            "type": "main",
            "index": 1
          }
        ]
      ]
    }
  }
}
```

**Comment by lublak** [comment]
i figure it out. Its a combination of empty data and merge node:

```json
{
  "name": "test",
  "nodes": [
    {
      "parameters": {},
      "name": "Start",
      "type": "n8n-nodes-base.start",
      "typeVersion": 1,
      "position": [
        470,
        200
      ]
    },
    {
      "parameters": {
        "values": {
          "number": [
            {
              "name": "number0"
            }
          ],
          "string": [
            {
              "name": "key",
              "value": "a"
            }
          ]
        },
        "options": {}
      },
      "name": "Set",
      "type": "n8n-nodes-base.set",
      "typeVersion": 1,
      "position": [
        780,
        200
      ]
    },
    {
      "parameters": {
        "values": {
          "number": [
            {
              "name": "number1",
              "value": 1
            }
          ],
          "string": [
            {
              "name": "key",
              "value": "a"
            }
          ]
        },
        "options": {}
      },
      "name": "Set1",
      "type": "n8n-nodes-base.set",
      "typeVersion": 1,
      "position": [
        780,
        350
      ]
    },
    {
      "parameters": {
        "values": {
          "number": [
            {
              "name": "number2",
              "value": 2
            }
          ],
          "string": [
            {
              "name": "key",
              "value": "a"
            }
          ]
        },
        "options": {}
      },
      "name": "Set2",
      "type": "n8n-nodes-base.set",
      "typeVersion": 1,
      "position": [
        750,
        500
      ]
    },
    {
      "parameters": {
        "mode": "mergeByKey",
        "propertyName1": "key",
        "propertyName2": "key"
      },
      "name": "Merge",
      "type": "n8n-nodes-base.merge",
      "typeVersion": 1,
      "position": [
        1030,
        350
      ]
    },
    {
      "parameters": {
        "mode": "mergeByKey",
        "propertyName1": "key",
        "propertyName2": "key"
      },
      "name": "Merge1",
      "type": "n8n-nodes-base.merge",
      "typeVersion": 1,
      "position": [
        1230,
        200
      ]
    },
    {
      "parameters": {
        "conditions": {
          "boolean": [
            {
              "value2": true
            }
          ]
        }
      },
      "name": "IF",
      "type": "n8n-nodes-base.if",
      "typeVersion": 1,
      "position": [
        890,
        500
      ]
    }
  ],
  "connections": {
    "Set": {
      "main": [
        [
          {
            "node": "Merge1",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "Set1": {
      "main": [
        [
          {
            "node": "Merge",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "Set2": {
      "main": [
        [
          {
            "node": "IF",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "Merge": {
      "main": [
        [
          {
            "node": "Merge1",
            "type": "main",
            "index": 1
          }
        ]
      ]
    },
    "Start": {
      "main": [
        [
          {
            "node": "Set",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "IF": {
      "main": [
        [
          {
            "node": "Merge",
            "type": "main",
            "index": 1
          }
        ]
      ]
    }
  },
  "active": false,
  "settings": {},
  "id": "23"
}
```

**Comment by janober** [comment]
Thanks a lot. Changed now the behaviour to be more consistent. As it would also execute if the second output does not have any input connection.

Will be released with the next version.

**Comment by lublak** [comment]
@janober thanks 👍

**Comment by janober** [comment]
Fix got released with n8n@0.101.0
